testing/local-binder-local-hub/binderhub_config.py

```diff
@@ -0,0 +1,111 @@
+######################################################################
+## A development config to test BinderHub locally.
+#
+# Run `python3 -m binderhub -f binderhub_config.py` in one terminal
+# Run `jupyterhub --config=binderhub_config.py` in another terminal
+
+# Host IP is needed in a few places
+import socket
+s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+s.connect(("8.8.8.8", 80))
+hostip = s.getsockname()[0]
+s.close()
+
+
+######################################################################
+## BinderHub config
+
+from binderhub.build_local import LocalRepo2dockerBuild
+
+
+c.BinderHub.debug = True
+c.BinderHub.use_registry = False
+c.BinderHub.builder_required = False
+
+c.BinderHub.build_class = LocalRepo2dockerBuild
+c.BinderHub.push_secret = None
+
+c.BinderHub.about_message = "<blink>Hello world.</blink>"
+c.BinderHub.banner_message = 'This is headline <a href="#">news.</a>'
+
+c.BinderHub.hub_url = f'http://{hostip}:8000'
+
+# Shared with JupyterHub
+api_token = "secretsecretsecretsecretsecretsecret"
+c.BinderHub.hub_api_token = api_token
+
+
+
+######################################################################
+## JupyterHub config
+from tornado import web
+
+cors = {}
+auth_enabled = False
+
+from dockerspawner import DockerSpawner
+
+# image & token are set via spawn options
+class LocalContainerSpawner(DockerSpawner):
```

Comment:
I'd like to move this into a mixin too.

Reply:
I tried... and failed .... https://github.com/jupyterhub/binderhub/pull/1358